main.rb

```diff
@@ -0,0 +1,177 @@
+# Enumerable Module extension
+module Enumerable
+  alias my_reduce my_inject
+
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_inject(carry = self[0])
+    i = carry == self[0] ? 1 : 0
+    while i < length
+      carry = yield(carry, self[i])
+      i += 1
+    end
+    carry
+  end
+
+  def my_select
+    selected = []
+    my_each do |i|
+      selected << i if yield(i)
+    end
+    selected
+  end
+
+  def my_all?
+    my_inject(true) { |res, i| res && yield(i) }
+  end
+
+  def my_any?
+    my_inject(false) { |res, i| res || yield(i) }
+  end
+
+  def my_none?
+    !my_any? { |i| yield(i) }
```

Comment:
What would happen if you delete the my_any? method, my_none wouldn't work either.